src/interfaces/IConditionalOrder.sol

```diff
@@ -11,7 +11,8 @@ import {IERC165} from "safe/interfaces/IERC165.sol";
  */
 interface IConditionalOrder {
     /// @dev This error is returned by the `getTradeableOrder` function if the order condition is not met.
-    error OrderNotValid();
+    ///      A parameter of `bytes4` type is included to allow the caller to specify the reason for the failure.
+    error OrderNotValid(bytes4);
```

Comment:
Feel free to ignore, but given this is only called in view functions do we care about the gas efficiency? How would we easily map a bytes4 to an error type when debugging (do we need a table for that)? 

Wouldn't it be more convenient to just log a string?

Reply:
This is fair. I was both optimising for gas and also after a way to ensure strong typings on error messages so that typos in tests are discouraged. Instead I broke out the error messages as constants to eliminate the typo problem. Best of both worlds!